Use eslint v9 flat config

plbstl opened the pull request:
## Priority*

- [x] High: This PR needs to be merged first for other tasks.
- [ ] Middle: This PR should be merged quickly to prevent conflicts due to common changes. (default)
- [ ] Low: This PR does not affect other tasks, so it can be merged later.

## Purpose of the PR*

Use eslint version 9 flat config.

## Changes*

This PR upgrades `eslint` to version 9 and uses the flat config.

## How to check the feature

Add code that warns or errors when `eslint` is set up correctly.


## Reference

- Closes #465

plbstl commented on eslint.config.mjs:
```diff
@@ -0,0 +1,63 @@
+  // Add compatibility
+  // @ts-expect-error - mismatched typings
+  ...fixupConfigRules(new FlatCompat().extends('plugin:react-hooks/recommended')),
+  // ...fixupConfigRules(new FlatCompat().extends('plugin:react-hooks/recommended', 'plugin:import/recommended')),
```
There is one issue I'm facing, I cannot get `eslint-plugin-import` to work properly

PatrykKuniczak replied:
It needs to work, because it's crucial 😄

plbstl replied:
The latest change includes a fix for this. 
Also noticed that the `eslint-plugin-import` configuration in the legacy config only worked with javascript files.

PatrykKuniczak commented on eslint.config.mjs:
Are you able to do it with `.mts`?
Because we want to have 100% `.ts` coverage in the near future

plbstl replied:
This is currently an experimental feature. https://eslint.org/blog/2024/08/eslint-v9.9.0-released/#experimental-typescript-configuration-files. It may be stable in the near future

PatrykKuniczak replied:
@plbstl If it works, go ahead :)

plbstl replied:
This works but for the CLI only, the vscode editor extension does not pick it up. I opened https://github.com/microsoft/vscode-eslint/issues/1927

plbstl replied:
It works when this is added to settings.json

```json
 "eslint.options": {
    "flags": ["unstable_ts_config"]
  }
```

PatrykKuniczak replied:
> It works when this is added to settings.json
> 
> ```json
>  "eslint.options": {
>     "flags": ["unstable_ts_config"]
>   }
> ```

Good, but it need to work for all IDE's :)

plbstl replied:
Maybe a section to tell what needs to be done since that would be on a per editor basis. https://eslint.org/docs/latest/use/integrations#editors

For sublime: the CLI flag is inputted in the `args` eslint setting. See https://github.com/SublimeLinter/SublimeLinter-eslint#examples, https://www.sublimelinter.com/en/master/linter_settings.html#args.
Same with JetBrains IDEs. See step 5 of https://www.jetbrains.com/help/idea/eslint.html#ws_js_eslint_manual_configuration

Basically passing in `--flag unstable_ts_config` when loading eslint would suffice for most.

PatrykKuniczak replied:
@plbstl In that case, leave this flag, it's the most certain solution

PatrykKuniczak commented:
Let's remove also:
```ts
// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-ignore
```

from `packages/i18n` and 
```ts
// eslint-disable-next-line
// @ts-ignore
```
from `pages/content-runtime/lib`

and check all `// eslint-disable-next-line` e.g. for `refresh` and `reload` because the most of it isn't necessary probably 😄

plbstl commented:
Okay, I'll make those changes now

PatrykKuniczak commented:
@plbstl Let's rebase ASAP, you'll avoid more conflicts, because i'm working on other new things 😄

plbstl commented:
For the recent changes in #697, I was thinking `pnpm exec` may be better than `pnpx` (pnpm dlx) since it uses the locally installed packages.

https://pnpm.io/cli/exec#examples

PatrykKuniczak commented on package.json:
```diff
@@ -8,7 +8,7 @@
   },
   "scripts": {
-    "clean:node_modules": "pnpx rimraf --glob **/node_modules",
+    "clean:node_modules": "rimraf --glob **/node_modules",
```
This `pnpx` is here with reason, read description:
https://github.com/Jonghakseo/chrome-extension-boilerplate-react-vite/pull/697#issue-2502937642

plbstl replied:
Okay

PatrykKuniczak commented:
@plbstl Are you able to set this `--flag unstable_ts_config` inside config file, instead of the flag in each script?

plbstl commented:
I checked the source code, the option is only available as a CLI flag

plbstl commented:
Replaced [`eslint-plugin-import`](https://github.com/import-js/eslint-plugin-import) with [`eslint-plugin-import-x`](https://github.com/un-ts/eslint-plugin-import-x) as it supports eslint v9.

It also has an additional rule [`no-rename-default`](https://github.com/un-ts/eslint-plugin-import-x/blob/master/docs/rules/no-rename-default.md) under [Helpful warnings](https://github.com/un-ts/eslint-plugin-import-x#helpful-warnings) section.

PatrykKuniczak commented:
@plbstl You've merge main wrongly, because i see unnecessary files(which shouldn't be shown as change here) in `Files changed`

And you have a lot of conflicts, let's solve it

plbstl commented:
@PatrykKuniczak The conflicts have now been resolved

PatrykKuniczak commented:
It doesn't work at all
![image](https://github.com/user-attachments/assets/a9985371-43b4-44b5-aa35-6ad2aef5edde)

First of all, eslint config wasn't found be IDE
Second thing, when i've added 
tsconfig for this eslint:
```json
{
  "$schema": "https://json.schemastore.org/tsconfig",
  "display": "Chrome Extension Utils",
  "extends": "./packages/tsconfig/base.json",
  "compilerOptions": {
    "noEmit": false
  },
  "include": ["eslint.config.ts"]
}
```
and "ready": "tsc -p eslint.tsconfig.json && turbo ready" for `package.json` and i've run it.
But it still doesn't work because of the lack of types.

And after all we have this warns, which are the problem, why we not decide to bump `eslint` to version 9:
![image](https://github.com/user-attachments/assets/8a2db37b-51fb-4ca2-a586-43fef35e6ac4)

I think this PR is overkill for that simply solution as eslint is.
IDN what to do with it, maybe @Jonghakseo have sth to say about it.
IMO the best idea is wait a little, when this libs start to support eslint 9

plbstl commented:
Can you tell me the name of the IDE? The repo does not contain any config files for IDEs. You have to instruct the specific IDE eslint plugin to load the config. Just as discussed in https://github.com/Jonghakseo/chrome-extension-boilerplate-react-vite/pull/699?notification_referrer_id=NT_kwDOAuvH57QxMjIzNjkxMTQ2MTo0OTAwNjU2Nw#discussion_r1747710475 

`@typescript-eslint/eslint-plugin` and `@typescript-eslint/parser` are no longer needed, a mistake on my part for not removing the dependencies. See https://typescript-eslint.io/packages/typescript-eslint/#migrating-from-legacy-config-setups

Same goes for `eslint-plugin-jsx-a11y`. Updating `eslint-plugin-jsx-a11y` to `6.10` fixes the peer dependency issue.

I don't know where the `eslint-plugin-import` issue is coming from. `eslint-import-resolver-typescript` has correct peer dependencies (https://github.com/import-js/eslint-import-resolver-typescript/blob/01902942c42292ab3cd266c2d6855c76a2ab5d7e/package.json#L64) and `eslint-plugin-import-x` is used and it is not depending on `eslint-plugin-import`.

Since `eslint-config-airbnb-typescript` is archived (https://github.com/iamturns/eslint-config-airbnb-typescript), I don't think it'll ever support version 9 or even update its peer dependencies.

PatrykKuniczak commented:
> Can you tell me the name of the IDE? The repo does not contain any config files for IDEs. You have to instruct the specific IDE eslint plugin to load the config. Just as discussed in #699
> 
> `@typescript-eslint/eslint-plugin` and `@typescript-eslint/parser` are no longer needed, a mistake on my part for not removing the dependencies. See https://typescript-eslint.io/packages/typescript-eslint/#migrating-from-legacy-config-setups
> 
> Same goes for `eslint-plugin-jsx-a11y`. Updating `eslint-plugin-jsx-a11y` to `6.10` fixes the peer dependency issue.
> 
> I don't know where the `eslint-plugin-import` issue is coming from. `eslint-import-resolver-typescript` has correct peer dependencies (https://github.com/import-js/eslint-import-resolver-typescript/blob/01902942c42292ab3cd266c2d6855c76a2ab5d7e/package.json#L64) and `eslint-plugin-import-x` is used and it is not depending on `eslint-plugin-import`.
> 
> Since `eslint-config-airbnb-typescript` is archived (https://github.com/iamturns/eslint-config-airbnb-typescript), I don't think it'll ever support version 9 or even update its peer dependencies.

You've pinned This PR in `Just as discussed in https://github.com/Jonghakseo/chrome-extension-boilerplate-react-vite/pull/699` it's correct?

I have been using WebStorm

plbstl commented:
> You've pinned This PR in `Just as discussed in https://github.com/Jonghakseo/chrome-extension-boilerplate-react-vite/pull/699` it's correct?

This is what I meant to link to: `Just as discussed in https://github.com/Jonghakseo/chrome-extension-boilerplate-react-vite/pull/699#discussion_r1747710475`

PatrykKuniczak commented:
@plbstl In the future, you can `Copy link`
![image](https://github.com/user-attachments/assets/97525d0a-23f7-470c-bca3-62a1d7bf9f92)

It's easier to navigate by everybody, because it creates clickable link 😄

PatrykKuniczak commented:
@plbstl Let's rebase branch

plbstl commented:
> @plbstl In the future, you can `Copy link` ![image](https://private-user-images.githubusercontent.com/64608510/366009323-97525d0a-23f7-470c-bca3-62a1d7bf9f92.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU5NjU2MjksIm5iZiI6MTcyNTk2NTMyOSwicGF0aCI6Ii82NDYwODUxMC8zNjYwMDkzMjMtOTc1MjVkMGEtMjNmNy00NzBjLWJjYTMtNjJhMWQ3YmY5ZjkyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTEwVDEwNDg0OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWYzYmQwOTEyZWI4YmNjYzkzMjE0YmZmMWM5N2QxNzY3ZGFhYWM4NTAzOWQzZGZiYzU1NDRkMTBlODE3NjZiNDUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.SslC6fO-i_62T6IkgCRiDfxogcP7deyJXCdvljDxMPA)
> 
> It's easier to navigate by everybody, because it creates clickable link 😄

Thank you, that's what I did but for some reason it added a search param for a `notificationId` before the URL hash

PatrykKuniczak reviewed (commented):
All your imports:
`import * as anything from "anything"`

should be:
`import anything from "anything"`

If it doesn't work you don't `include` or set `types` in tsconfig properly

plbstl commented:
> All your imports: `import * as anything from "anything"`
> 
> should be: `import anything from "anything"`
> 
> If it doesn't work you don't `include` or set `types` in tsconfig properly

It works. Its just that eslint shows a warning for the [`import-x/no-named-as-default-member`](https://github.com/un-ts/eslint-plugin-import-x/blob/docs/rules/no-named-as-default-member.md) rule.

Should I change it back and turn off the rule? Or change the imports to named imports? Or something else?

PatrykKuniczak commented:
> > All your imports: `import * as anything from "anything"`
> > should be: `import anything from "anything"`
> > If it doesn't work you don't `include` or set `types` in tsconfig properly
> 
> It works. Its just that eslint shows a warning for the [`import-x/no-named-as-default-member`](https://github.com/un-ts/eslint-plugin-import-x/blob/docs/rules/no-named-as-default-member.md) rule.
> 
> Should I change it back and turn off the rule? Or change the imports to named imports? Or something else?

Maybe set it off, because we're using default imports

PatrykKuniczak commented:
@plbstl You're right, because no default is exported from there :)
![image](https://github.com/user-attachments/assets/1cefb55d-7cc6-4ab4-b7cf-76ec69d11887)

Leave it, in the way you've done it

But there's greater problem. my IDE still can't find config, because of `.ts` i think this should be `.js` because it works.

Or use `.ts` but build this file on `postinstall` and IDE will be using bundled version of this, let's check if it will be working 😄

plbstl commented:
> But there's greater problem. my IDE still can't find config, because of `.ts` i think this should be `.js` because it works.
> 
> Or use `.ts` but build this file on `postinstall` and IDE will be using bundled version of this, let's check if it will be working 😄
> 
> If yes, then add `.js` to .gitignore and it will work perfect :)
> 
> PS: With second approach, we can have 100% ts coverage, maybe it's worth efford 😄

The latest changes have done exactly that, and it works well.

PatrykKuniczak commented:
@plbstl I'll check it tomorrow morning 😸

PatrykKuniczak commented on eslint.config.ts:
I have this error:
![image](https://github.com/user-attachments/assets/43de085a-2ece-4efa-b328-a4ffdcc71738)

For those imports:
![image](https://github.com/user-attachments/assets/3b37b905-b864-4520-9a08-16f26f4b4f5c)

plbstl replied:
A fix has been implemented for this

PatrykKuniczak reviewed (changes requested):
Remove all `--flag unstable_ts_config` because now we have `.js` and we don't need to take a risk with `unstable` things

plbstl commented on package.json:
```diff
@@ -36,34 +37,40 @@
     "react-dom": "18.3.1"
   },
   "devDependencies": {
+    "@eslint/compat": "^1.1.1",
+    "@eslint/eslintrc": "^3.1.0",
+    "@eslint/js": "^9.10.0",
```
It may be better to leave it the default way as its more predictable and how most projects are structured.
This may not count but installing a new package re-sorts the dependencies alphabetically.

plbstl commented on package.json:
```diff
@@ -27,6 +27,7 @@
     "lint:fix": "turbo lint:fix --continue -- --fix --cache --cache-location node_modules/.cache/.eslintcache",
     "prettier": "turbo prettier --continue -- --cache --cache-location node_modules/.cache/.prettiercache",
     "prepare": "husky",
+    "postinstall": "esbuild eslint.config.ts --outfile=eslint.config.js && prettier eslint.config.js --write --ignore-path .prettierignore",
```
The latest changes implement this. But the name of the script is `build:eslint`

PatrykKuniczak commented on package.json:
```diff
@@ -16,6 +16,7 @@
     "clean:install": "pnpm clean:node_modules && pnpm install --frozen-lockfile",
     "build": "pnpm clean:bundle && turbo ready && turbo build",
     "build:firefox": "pnpm clean:bundle && turbo ready && cross-env __FIREFOX__=true turbo build",
+    "build:eslint": "esbuild eslint.config.ts --outfile=eslint.config.js",
```
It this case, when you have `tsconfig` you don't really need `esbuild` you can build it via `tsc`
I will probably remove all `esbuild` from entire project to simplify it, because it's harder and harder to maintain.

Let me know

plbstl replied:
Implemented this in the latest change

PatrykKuniczak commented:
@plbstl Rebase and resolve conflicts

PatrykKuniczak commented on package.json:
```diff
@@ -16,6 +16,7 @@
     "clean:install": "pnpm clean:node_modules && pnpm install --frozen-lockfile",
     "build": "pnpm clean:bundle && turbo ready && turbo build",
     "build:firefox": "pnpm clean:bundle && turbo ready && cross-env __FIREFOX__=true turbo build",
+    "build:eslint": "tsc --build",
```
```suggestion
    "build:eslint": "tsc -p eslint.json",
```

IMO this should looks like this, because `tsconfig.json` isn't necessary,
but include .ts config inside `eslint.json`

IMO 2 configs for that simple task is overkill, becasue you won't have other files in root, like we have in packages

plbstl replied:
It was initially only `tsconfig.json`, I then added the `@extension/tsconfig/eslint.json` with the latest changes to keep all tsconfigs in one place.

> IMO 2 configs for that simple task is overkill, becasue you won't have other files in root, like we have in packages

What do you suggest?

PatrykKuniczak replied:
@plbstl Your current solution is good for now.
i'll refactor it a little, if i'll be able, on my branch.

Good work <3

PatrykKuniczak replied:
It works the same without this flag, in other packages we doesn't use it.
![image](https://github.com/user-attachments/assets/db354f53-4482-42b0-9ea0-5886e8c1b54c)

It's really necessary?

plbstl replied:
Not necessarily, for this use case. See https://www.typescriptlang.org/docs/handbook/project-references.html#caveats-for-project-references

PatrykKuniczak replied:
@plbstl Ok, let's leave it here, it's seems nice option.
I'll use it for all `tsc` :)

plbstl commented:
I have merged in the main branch, but I did not run into merge conflicts

PatrykKuniczak commented:
@plbstl 
I have errors still:
![image](https://github.com/user-attachments/assets/d0a3ce04-b37e-4a39-b3d9-5bdf1c60f14a)

Because this packages doesn't export default nothing:
![image](https://github.com/user-attachments/assets/997ca49a-f314-45a1-aeae-c62a2da2df01)

Let's change it to:
import * as name from 'name';

It's working :)

plbstl commented:
> @plbstl 
> 
> I have errors still:
> 
> ![image](https://github.com/user-attachments/assets/d0a3ce04-b37e-4a39-b3d9-5bdf1c60f14a)
> 
> 
> 
> Because this packages doesn't export default nothing:
> 
> ![image](https://github.com/user-attachments/assets/997ca49a-f314-45a1-aeae-c62a2da2df01)
> 
> 
> 
> Let's change it to:
> 
> `import * as name from 'name';`
> 
> 
> 
> It's working :)

I think it's because the root tsconfig was removed. It has to be added back since it includes configuration for eslint.config.ts

VSCode does not error for that, it made me totally forget about it

PatrykKuniczak commented:
@plbstl Yeah, the tsconfig are using also for this, but anyway i'm thinking if it's good, but yeah, do it like you said, but extends `base.json`

plbstl commented:
I have made the relevant changes

PatrykKuniczak commented:
@plbstl 
![image](https://github.com/user-attachments/assets/4ce835f3-280f-458d-a5bf-03cab7bf7df8)
Why? xD

Let's rebase locally and see those conflicts, if there's no conflicts, then IDN xD

PatrykKuniczak reviewed (approved):
LGTM 

<img src="https://media4.giphy.com/media/QyrysGo7Hz70I/giphy.gif"/>

PatrykKuniczak reviewed (changes requested):
![image](https://github.com/user-attachments/assets/a3ec0ad2-2861-4454-807f-de52b3fa8ea5)

You did sth wrong for sure, let's rebase it and i'll try it again :)

plbstl commented:
> ![image](https://private-user-images.githubusercontent.com/64608510/367523267-a3ec0ad2-2861-4454-807f-de52b3fa8ea5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjYzNDIzODMsIm5iZiI6MTcyNjM0MjA4MywicGF0aCI6Ii82NDYwODUxMC8zNjc1MjMyNjctYTNlYzBhZDItMjg2MS00NDU0LTgwN2YtZGU1MmIzZmE4ZWE1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTE0VDE5MjgwM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTEwMDRlNWNmZTM1ZDU3NTRlMmU4YTQwMzMyN2Q3ZmNjN2VjN2RhNDZkZmM3MTc4OTIxYjJkOGQzMTEzZmU5YmImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.j4-yCfSBKIFKa5BCNF5T7kVGx_I5dplGII3usNuETos)
> 
> You did sth wrong for sure, let's rebase it and i'll try it again :)

- The `@ts-ignore` pragma was removed from `packages/i18n/index.ts` as per https://github.com/Jonghakseo/chrome-extension-boilerplate-react-vite/pull/699#issuecomment-2328860784.
- The file that is being imported is generated after a `pnpm build` or `pnpm dev`.
- The Action runs `pnpm type-check && pnpm build`. type-check is run before the file is generated. 

It can be changed that build is run first. Or add `@ts-ignore` back.

PatrykKuniczak commented:
OOOKKK, that's what i'm talking about with @Jonghakseo 
I'll open PR, to solve this

PatrykKuniczak commented:
It'll be fixed in #740 
After this, everything should be OK.
Thanks again for your contribution <3

PatrykKuniczak commented:
@plbstl If you'll eager to contribute a little more visit our [Discord](https://discord.com/invite/4ERQ6jgV9a) and i can explain you more 😄

plbstl commented:
@PatrykKuniczak yeah sure. I just joined the server 🚀

PatrykKuniczak reviewed (changes requested):
I've resolve conflicts, and add 1 missing types, but theres still some work:
As you can see, there's a bunch of warns:
![image](https://github.com/user-attachments/assets/43dd9731-feab-4ea7-9990-23e0e5e322a1)

Let's fix it, but:
https://github.com/iamturns/eslint-config-airbnb-typescript
Isn't longer supported, you need to find sth else, which supporting eslint v9
`eslint-config-airbnb-base` is also have last update 3 years ago
`eslint-plugin-import` is still as a peer deps
And there's missing types:
![image](https://github.com/user-attachments/assets/5a080da2-ea94-4e04-b493-d65f8d3ccd89)

Let's try to find solution, because we don't want to merge it, becasue of potential problems with deps, we want to avoid it, that's why we don't decided to implement it for now, but i hope you're able to fix it.

PS: For reproduce that error try e.g. `pnpm add -D nodemon -w`

PatrykKuniczak commented:
@plbstl Are you eager to go on?

plbstl commented:
Apologies for the extended delay on this PR. I’m back now and ready to provide updates and address any pending feedback.

Here’s a quick update on the current status:

- The `eslint-config-airbnb` package is still in the process of supporting v9 (https://github.com/airbnb/javascript/issues/2961).
- An alternative to [`iamturns/eslint-config-airbnb-typescript`](https://github.com/iamturns/eslint-config-airbnb-typescript) is [`Kenneth-Sills/eslint-config-airbnb-typescript`](https://github.com/Kenneth-Sills/eslint-config-airbnb-typescript), but the `eslint-config-airbnb` issue above, is a blocker.
- Both `eslint-config-prettier` and `eslint-plugin-prettier` are currently listed as dependencies, but only `eslint-plugin-prettier` was/is used.

**Observing the config file, `eslint-config-airbnb` was not used in the old eslint config and is still not being used in the flat config.**

PatrykKuniczak commented:
@plbstl If `eslint-config-airbnb` isn't necessary, let's remove it and you'll be a step ahead.

For the future, let's update branch with `rebase` instead of `merge` :)

plbstl commented:
> @plbstl If `eslint-config-airbnb` isn't necessary, let's remove it and you'll be a step ahead.

Okay, I have removed `eslint-config-airbnb-typescript` along with `eslint-config-prettier`.

> For the future, let's update branch with `rebase` instead of `merge` :)

Duely noted, I have now updated the branch with `rebase` :)

---

There are no `pnpm install` warnings anymore, and the missing typedefs warnings in `eslint.config.ts` are also resolved when opened in VSCode. I currently don't have access to WebStorm to confirm.

PatrykKuniczak commented on package.json:
```diff
@@ -16,6 +16,7 @@
     "clean:install": "pnpm clean:node_modules && pnpm install --frozen-lockfile",
     "build": "pnpm clean:bundle && turbo ready && turbo build",
     "build:firefox": "pnpm clean:bundle && turbo ready && cross-env __FIREFOX__=true turbo build",
+    "build:eslint": "tsc --build",
```
```suggestion
    "build:eslint": "tsc -b",
```

To match other modules from `dev` branch :)

PatrykKuniczak reviewed (changes requested):
I have warning:
![image](https://github.com/user-attachments/assets/a6831604-7f07-4e8f-8318-6d5b120ab857)

Try to avoid it

PatrykKuniczak commented:
@plbstl Also tests failed, let's take a closer look on that.

plbstl commented:
ESLint should no longer display a warning for an unspecified React version. An update to a deeper dependency is causing the e2e tests to fail. This is useful to note in case future test failures occur without related code changes.

PatrykKuniczak reviewed (approved):
LGTM

Very good job ❤️
